Lib/test/test_exceptions.py

```diff
@@ -410,6 +410,32 @@ def test_WindowsError(self):
             self.assertEqual(w.strerror, 'foo')
             self.assertEqual(w.filename, None)
             self.assertEqual(w.filename2, None)
+            self.assertEqual(str(w), "[Errno bar] foo")
+
+            # Unexpected number of arguments
+            w = OSError(0)
+            self.assertEqual(w.errno, None)
+            self.assertEqual(w.winerror, None)
+            self.assertEqual(w.strerror, None)
+            self.assertEqual(w.filename, None)
+            self.assertEqual(w.filename2, None)
+            self.assertEqual(str(w), "0")
+
+            w = OSError('foo')
+            self.assertEqual(w.errno, None)
+            self.assertEqual(w.winerror, None)
+            self.assertEqual(w.strerror, None)
+            self.assertEqual(w.filename, None)
+            self.assertEqual(w.filename2, None)
+            self.assertEqual(str(w), "foo")
+
+            w = OSError('a', 'b', 'c', 'd', 'e', 'f')
+            self.assertEqual(w.errno, None)
+            self.assertEqual(w.winerror, None)
+            self.assertEqual(w.strerror, None)
+            self.assertEqual(w.filename, None)
+            self.assertEqual(w.filename2, None)
+            self.assertEqual(str(w), "('a', 'b', 'c', 'd', 'e', 'f')")
```

Comment:
if this our own test or copied from cpython? we usually don't write our own tests here but in `extra_tests/snippets/`

Reply:
Moved the new tests to `extra_tests/snippets/builtin_exceptions.py`